### PR TITLE
[DST-358]: Exposing Listbox component

### PR DESCRIPTION
**Pull request by aromko**
- we expose the listbox component
- the component documentation for listbox is added
- a recipe for using listbox is added

### CODE REVIEW

**Comment by sebald** [comment]
@aromko @OsamaAbdellateef Deselecting an item seems not to work.

[Kapture 2024-02-13 at 14.04.28.webm](https://github.com/marigold-ui/marigold/assets/985701/2c379696-caba-4df4-9bd9-493217038f4e)

**Comment by OsamaAbdellateef** [comment]
> @aromko @OsamaAbdellateef Deselecting an item seems not to work.
> 
>  Kapture.2024-02-13.at.14.04.28.webm
the conflict comes from the styling, 
focus and selected have the same effect in core theme 
if you tried this with b2b theme you will notice that it works because focus and selected have the same effect  @sebald

**Comment by sebald** [comment]
@OsamaAbdellateef ah I see, so I guess we need to adjust the stlyes 👀 Otherwise this is super weird

**Comment by sarahgm** [comment]
@OsamaAbdellateef @sebald @aromko is this caused by this you mean? 
https://reservix.atlassian.net/jira/software/projects/DST/boards/134?selectedIssue=DST-280

**Comment by OsamaAbdellateef** [comment]
> @OsamaAbdellateef @sebald @aromko is this caused by this you mean? https://reservix.atlassian.net/jira/software/projects/DST/boards/134?selectedIssue=DST-280

Yes it is :100: @sarahgm

File: docs/content/recipes/listbox-recipe.mdx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
```
[the remark this answers is not in the document]
Comment (aromko) [reply docs/content/recipes/listbox-recipe.mdx:1]: You don't see the content from listbox-recipe.mdx?

**Pull request closed without merging**